labels/validate.go

```diff
@@ -0,0 +1,19 @@
+package labels
+
+import (
+	"github.com/containerd/containerd/errdefs"
+	"github.com/pkg/errors"
+)
+
+const (
+	maxSize = 4096
+)
+
+func Validate(k, v string) error {
+	// A label key and value should be under 4096 bytes
+	if (len(k) + len(v)) > maxSize {
+		return errors.Wrapf(errdefs.ErrInvalidArgument, "label key and value greater than maximum size (%d bytes), key: %v", maxSize, k[0:10])
```

Comment:
Make sure that `k` is greater than 10 or this will panic. Use `%10s` to limit the length in the format string (I think that is the syntax).